common/src/backends/otlp_tracing_bridge.rs

```diff
@@ -0,0 +1,154 @@
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_f64(&mut self, field: &tracing::field::Field, value: f64) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_i64(&mut self, field: &tracing::field::Field, value: i64) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    // TODO: Remaining field types from AnyValue : Bytes, ListAny, Boolean
+}
+
+pub struct OpenTelemetryTracingBridge<P, L>
+where
+    P: LoggerProvider<Logger = L> + Send + Sync,
+    L: Logger + Send + Sync,
+{
+    logger: L,
+    _phantom: std::marker::PhantomData<P>, // P is not used.
+}
+
+impl<P, L> OpenTelemetryTracingBridge<P, L>
+where
+    P: LoggerProvider<Logger = L> + Send + Sync,
+    L: Logger + Send + Sync,
+{
+    pub fn new(provider: &P) -> Self {
+        OpenTelemetryTracingBridge {
+            logger: provider.versioned_logger(
+                INSTRUMENTATION_LIBRARY_NAME,
+                Some(Cow::Borrowed(env!("CARGO_PKG_VERSION"))),
+                None,
+                None,
+            ),
+            _phantom: Default::default(),
+        }
+    }
+}
+
+impl<S, P, L> Layer<S> for OpenTelemetryTracingBridge<P, L>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    P: LoggerProvider<Logger = L> + Send + Sync + 'static,
+    L: Logger + Send + Sync + 'static,
+{
+    fn on_event(&self, event: &tracing::Event<'_>, ctx: tracing_subscriber::layer::Context<'_, S>) {
+        let meta = event.metadata();
+        let mut log_record: LogRecord = LogRecord::default();
+        log_record.severity_number = Some(severity_of_level(meta.level()));
+        log_record.severity_text = Some(meta.level().to_string().into());
+
+        // Extract the trace_id & span_id from the opentelemetry extension.
+        if let Some((trace_id, span_id)) = ctx.lookup_current().and_then(|span| {
+            span.extensions()
+                .get::<OtelData>()
+                .and_then(|ext| ext.builder.trace_id.zip(ext.builder.span_id))
+        }) {
+            log_record.trace_context = Some(TraceContext::from(&SpanContext::new(
+                trace_id,
+                span_id,
+                TraceFlags::default(),
+                false,
+                TraceState::default(),
+            )));
```

Comment:
Are the logs attached to the spans (visible in datadog) because of attaching this `trace_context` to them, which contains the `trace_id` and `span_id`?